backend/de.metas.business/src/main/java-legacy/org/compiere/model/MProduct.java

```diff
@@ -332,10 +334,40 @@ protected boolean afterSave(boolean newRecord, boolean success)
 			}
 			insert_Tree(X_AD_Tree.TREETYPE_Product);
 		}
+		// Product category changed, then update the accounts
+		if (!newRecord && is_ValueChanged(I_M_Product.COLUMNNAME_M_Product_Category_ID))
+		{
+			update_Accounting();
+		}
 		
 		return true;
 	}	// afterSave
-	
+
+	private void update_Accounting()
+	{
+		final String sql = DB.convertSqlToNative("UPDATE M_Product_Acct pa "
+				//
+				+ "SET (P_Revenue_Acct,P_Expense_Acct,P_CostAdjustment_Acct,P_InventoryClearing_Acct,P_Asset_Acct,P_COGS_Acct,"
+				+ " P_PurchasePriceVariance_Acct,P_InvoicePriceVariance_Acct,"
+				+ " P_TradeDiscountRec_Acct,P_TradeDiscountGrant_Acct,"
+				+ " P_WIP_Acct,P_FloorStock_Acct,P_MethodChangeVariance_Acct,P_UsageVariance_Acct,P_RateVariance_Acct,"
+				+ " P_MixVariance_Acct,P_Labor_Acct,P_Burden_Acct,P_CostOfProduction_Acct,P_OutsideProcessing_Acct,P_Overhead_Acct,P_Scrap_Acct)="
+				//
+				+ " (SELECT P_Revenue_Acct,P_Expense_Acct,P_CostAdjustment_Acct,P_InventoryClearing_Acct,P_Asset_Acct,P_COGS_Acct,"
+				+ " P_PurchasePriceVariance_Acct,P_InvoicePriceVariance_Acct,"
+				+ " P_TradeDiscountRec_Acct,P_TradeDiscountGrant_Acct,"
+				+ " P_WIP_Acct,P_FloorStock_Acct,P_MethodChangeVariance_Acct,P_UsageVariance_Acct,P_RateVariance_Acct,"
+				+ " P_MixVariance_Acct,P_Labor_Acct,P_Burden_Acct,P_CostOfProduction_Acct,P_OutsideProcessing_Acct,P_Overhead_Acct,P_Scrap_Acct"
+				+ " FROM M_Product_Category_Acct pca"
+				+ " WHERE pca.M_Product_Category_ID=" + getM_Product_Category_ID()
+				+ " AND pca.C_AcctSchema_ID=pa.C_AcctSchema_ID"
+				+ "), Updated=now(), UpdatedBy=0 "
+				//
+				+ "WHERE pa.M_Product_ID=" + getM_Product_ID()
+				+ " AND EXISTS (SELECT 1 FROM M_Product p WHERE p.M_Product_ID=pa.M_Product_ID AND p.M_Product_Category_ID=" + getM_Product_Category_ID() + ")");
+
+		DB.executeUpdateEx(sql, get_TrxName());
```

Comment:
Please extract to (in order of my preference)
- DAO method to perform update
- An SQL function (called from DAO method)
- DAO method to call this anonymous code.
and consider refactoring `org.compiere.process.ProductCategoryAcctCopy#doIt` to use the same approach.